docs/4-development/10-testing.md

```diff
@@ -444,3 +444,80 @@ yarn test:cypress
 # start a static server in the `coverage` folder and inspect the results in the browser
 http-server coverage
 ```
+
+## Guideliness
+
+Focus on testing behavior rather than tests that tightly couple with implementation details as they make refactoring difficult and to ensure your tests remain useful even as the codebase evolves. As a rule of thumb, focus on feature-centric tests asserting how a feature works as a whole and test the available user interactions.
+
+### Write Tests That
+
+- Verify the expected behavior of the component, not its internal workings.
+- Mimic real user interactions (clicks, typing, form submissions) instead of testing component internals.
+
+```tsx
+it("should focus input on click of the label", () => {
+  cy.mount(
+    <>
+      <Label for="form-ui5-input">Label for Input</Label>
+      <Input id="form-ui5-input"></Input>
+    </>
+  );
+
+  // act
+  cy.get("[ui5-label]").realClick();
+
+  // assert
+  cy.get("[ui5-input]").should("be.focused");
+});
+```
+
+```tsx
+it("Change event is not fired when the same suggestion item is selected (with typeahead)", () => {
+  // act
+  cy.get("@input")
+    .realClick();
+  cy.get("@input")
+    .realType("a");
+  cy.get("@input").realPress("Enter");
+
+  // assert
+  cy.get("@input").should("have.value", "Afghanistan");
+});
+```
+
+- Are independent (each test should stand alone and not rely on the outcome of another test).
+- Are fast and run efficiently to avoid slowing down development.
+
+
+### Avoid Tests That
+
+- Directly assert implementation details as minor refactors will require unnecessary test changes.
+
+```css
+:host {
+  display: block;
+  text-overflow: ellipsis;
+  overflow: hidden;
+}
+```
+
+```tsx
+cy.get("@myComponent")
+  .should("have.css", "display", "block")
+  .should("have.css", "overflow", "hidden")
+  .should("have.css", "text-overflow", "ellipsis");
+```
+
```

Comment:
this example seems like something that should be done. I think there  should be a **DON'T** and then provide a **DO** that shows measuring the height